scrapy/utils/url.py

```diff
@@ -98,7 +96,10 @@ def guess_scheme(url):
                     )?      # optional match of ".", ".." or ".blabla"
                     /       # at least one "/" for a file path,
                     .       # and something after the "/"
-                    ''', parts.path, flags=re.VERBOSE):
+                    ''', url, flags=re.VERBOSE):
+        return any_to_uri(url)
+    # Windows drive-letter path
+    elif re.match(r'''^[a-z]:\\''', url, flags=re.IGNORECASE):
```

Comment:
could you please add tests for these changes? I also wonder if these two conditions can be moved to helper functions, looks_like_posix_path and looks_like_windows_path; it may allow easier testing.